Loop/Managers/DeviceDataManager.swift

```diff
@@ -1357,84 +1366,74 @@ extension Notification.Name {
 }
 
 // MARK: - Remote Notification Handling
-extension DeviceDataManager {
+
+extension DeviceDataManager: RemoteActionDelegate {
     
     func handleRemoteNotification(_ notification: [String: AnyObject]) {
         Task {
-            let backgroundTask = await beginBackgroundTask(name: "Remote Data Upload")
-            await handleRemoteNotification(notification)
+            log.default("Remote Notification: Handling notification %{public}@", notification)
+            
+            guard FeatureFlags.remoteCommandsEnabled else {
+                log.error("Remote Notification: Remote Commands not enabled.")
+                return
+            }
+            
+            let backgroundTask = await beginBackgroundTask(name: "Handle Remote Notification")
+            do {
+                try await remoteDataServicesManager.handleRemoteNotification(notification)
+            } catch {
+                log.error("Remote Notification: Error: %{public}@", String(describing: error))
+            }
+            
             await endBackgroundTask(backgroundTask)
+            log.default("Remote Notification: Finished handling")
         }
     }
     
-    func handleRemoteNotification(_ notification: [String: AnyObject]) async {
-        
-        defer {
-            log.default("Remote Notification: Finished handling")
-        }
-        
-        guard FeatureFlags.remoteCommandsEnabled else {
-            log.error("Remote Notification: Remote Commands not enabled.")
-            return
-        }
-        
-        let command: RemoteCommand
-        do {
-            command = try await remoteDataServicesManager.commandFromPushNotification(notification)
-        } catch {
-            log.error("Remote Notification: Parse Error: %{public}@", String(describing: error))
-            return
+    func processPendingRemoteCommands() {
```

Comment:
As much as we can, "remote" handling responsibility/concerns should be isolated to RemoteDataServicesManager/ServicesManager. I'm not sure DeviceDataManager needs to change at all.